### DISCUSSION TITLE
Library Imports

### DISCUSSION THREAD

**Discussion by ElectronicsArchiver** [denoland/deno#16192] [Q&A] [answered]
<br>

I was wondering if there is any way of handling 
imports in modules / libraries similarly to the 
way you can do with import-maps in programs.

<br>

While I know of the deps.ts approach, the last 
time I looked into it, it was said to be bad 
practice due to everything accessing all the 
imports, not sure if that is ( still ) correct?

<br>

*5th day: Still no one?*

<br>

**Comment by zachauten** [comment, marked as answer]
> 5th day: Still no one?

Could you try to rephrase your question, maybe with a code example? I think it's a little unclear what you're asking.

> the last time I looked into it, it was said to be bad practice

Maybe you could provide a link to the discussion you're referencing, to give more clarity?

**Reply by ElectronicsArchiver** [reply]
<br>

I don't want to write 

```js
import { x } from 'longUrlWithVersion'
```

in my module / library and instead do
something like denos importmaps where
I can specify a name for the import url.

However obviously this can't be done if you don't 
control the `deno run` `--importmaps` parameter, 
nor would it work out in combination with other libraries.

<br>

> Maybe you could provide a link to the discussion you're referencing, to give more clarity?

You think I still have a link to that discussion, not to mention still know on what website it was ..

*I just referenced it, since it was the last thing I could remember.*

<br>

**Reply by zachauten** [reply]
I think you may be out of luck with import maps, but the [Deno manual does recommend using deps.ts](https://deno.land/manual@v1.26.1/linking_to_external_code#it-seems-unwieldy-to-import-urls-everywhere), so I don’t think that’s considered bad practice.
